packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -528,6 +575,7 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
    * should only be called from within a queued callback. This is necessary
    * because the queue shouldn't rely on another queued callback.
    */
+  // TODO: Find where this is called and see if we can run the middleware before it
```

Comment:
This is only used during initialization and then in a few select cases within this file. I need to do some thinking about the appropriate places in the init phase to use middleware. Off the top of my head, the only place init should run middleware is when signing in a user from redirect (i.e. first page load after a `signInWithRedirect()` call)

Reply:
Inserted it in what I think is the correct place in `initializeCurrentUser()`.